corehq/ex-submodules/casexml/apps/phone/restore.py

```diff
@@ -381,7 +478,13 @@ def get_payload(self):
 
     def get_response(self):
         try:
-            return HttpResponse(self.get_payload(), mimetype="text/xml")
+            payload = self.get_payload()
+            try:
+                f = open(payload, 'r')
+                # Since payload file is all one line, need to readline based on bytes
+                return StreamingHttpResponse(iter(lambda: f.readline(MAX_BYTES), ''), mimetype="text/xml")
+            except IOError:
```

Comment:
I think this is the only place where this idiom is used outside of test
cases. One option would be to do something like this:
```
try:
    with open(payload, 'r') as f:
        try:
            ....
        exception IOError:
            # do something
except IOError:
    # no such file or directory
```

Another option is to check the `errno` attrib of the exception which should
be [ENOENT](https://docs.python.org/2/library/errno.html#errno.ENOENT) if
the file does not exist. I'm not sure if that's a good practice to do.